### DISCUSSION TITLE
Does Prism support Azure implementation? (OpenAI, RAG, ...)

### DISCUSSION THREAD

**Discussion by VerheeckeLorenzo** [prism-php/prism#95] [Q&A] [answered]
As the title cites, I was wondering if the Prism package also supports connecting to an OpenAI text/embedding model hosted on Azure?

**Comment by sixlive** [comment]
It should! I just signed up for an Azure account. I'll test things out and circle back!

**Reply by sixlive** [reply]
Okay, I just kind of figured it was OpenAI API compatible, doesn't look like it is. I'll have to create an Azure provider. I'll create an issue for it but I might not get back to it for a few days.

**Reply by avengerweb** [reply]
Any updates on this one? it is compatible. 
Difference is, you need to have url with model name and "api-version" in query param. It is pretty small change, I think you should add config ('is_azure') in configuration

https://github.com/prism-php/prism/blob/main/src/Providers/OpenAI/Handlers/Text.php#L108
```php
return $this->client->post($request->model() . '/chat/completions?api-version=' . config('openai.azure_version'),
```

**Reply by VerheeckeLorenzo** [reply]
I think it would be best to keep those two seperated. Azure OpenAI might not be too different from OpenAI's official's implementation (API), but this could maybe change in the future, no? Also: Azure supports [RAG](https://azure.microsoft.com/en-us/products/ai-services/ai-search)/other cloud solutions that might be cool to implement in this package. Just sharing my thoughts here ^^

**Reply by avengerweb** [reply]
Even official SDK do not separate them. Even though there indeed could be more features, adding simple flag is much faster then implement whole provider from scratch.
Plus, API of azure has versioning support, it will not just suddenly changed.

**Reply by ChrisB-TL** [reply, marked as answer]
For the soon to be released Bedrock provider we've fallen somewhere between the two. 

For this, to future proof, it probably makes sense to have a separate provider with its own client and config but using the existing openai handlers - rather than a config flag. Not much extra effort.

**Reply by mafrasil** [reply]
Any developments about supporting Azure?

**Reply by imYashGupta** [reply]
any update on this?
